### PR TITLE
requests-presentation  added

### PR DESCRIPTION
**Pull request by cmigazzi**
This is my article on the requests library.

### CODE REVIEW

File: content/articles/presentation-requests.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+### Installation
+`pipenv install requests`
+
+### Une simple requête
+
+Pour réaliser une requête:
+
+avec `urllib3`...
+```
+import urllib3
```
Comment (tchappui) [inline content/articles/presentation-requests.md]: from urllib3 import PoolManager

File: content/articles/presentation-requests.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+http = PoolManager()
+res = http.request("GET", "https://fr.openfoodfacts.org/")
+```
+... et pour obtenir le même résultat avec `requests`:
+```
+import requests
+
+res = requests.get("https://fr.openfoodfacts.org/ ")
```
Comment (tchappui) [inline content/articles/presentation-requests.md]: pourquoi un espace à la fin de l'url?

File: content/articles/presentation-requests.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+```
+C'est effectivement beaucoup plus simple, mais que se passe-t-il ?
+
+Dans les 2 cas, `res` contient la réponse du serveur HTTP.
+
+Avec `requests`, l’import de `http` et l’instanciation de `PoolManager()` est transparent pour l’utilisateur, ce qui économise du code. Les méthodes HTTP sont des méthodes de `requests` au lieu d'être des paramètres dans `urllib3`.
```
Comment (tchappui) [inline content/articles/presentation-requests.md]: ce sont techniquement des fonctions

File: content/articles/presentation-requests.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+res = request.post("https://mon-url.de/post", data={key:value})
+res = request.put("https://mon-url.de/put", data={key:value})
+res = request.delete("https://mon-url.de/delete")
+res = request.head("https://mon-url.de/get")
+res = request.head("https://mon-url.de/get")
+```
+
+**Et qu'est-ce qu'on en fait ?**
+
+`requests` renvoie un objet `Response` qui possède les attributs suivants:
```
Comment (tchappui) [inline content/articles/presentation-requests.md]: je dirais objet de type Response, car res est une instance de réponse

File: content/articles/presentation-requests.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+>>> res.encoding
+'UTF-8'
+```
+Et quelques méthodes de base:
+
+```
+# le code html
+>>> res.text # chaine de charactère
+
+# Si la réponse est au format json, on peut la convertir en dictionnaire
```
Comment (tchappui) [inline content/articles/presentation-requests.md]: en dictionnaire ou en liste

Comment (cmigazzi) [reply content/articles/presentation-requests.md]: res.json() renvoie bien un dictionnaire

File: content/articles/presentation-requests.md (hunk 1 above)
Comment (cmigazzi) [reply content/articles/presentation-requests.md]: effectivement

File: content/articles/presentation-requests.md (hunk 2 above)
Comment (cmigazzi) [reply content/articles/presentation-requests.md]: erreur de frappe

File: content/articles/presentation-requests.md (hunk 3 above)
Comment (cmigazzi) [reply content/articles/presentation-requests.md]: c'est vrai

File: content/articles/presentation-requests.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
+Title: Présentation de la librairie requests
+Date: 2019-01-31 16:00
+Modified: 2019-01-31 16:00
```
[the remark this answers is not in the document]
Comment (cmigazzi) [reply content/articles/presentation-requests.md]: Ok

File: content/articles/presentation-requests.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
+Title: Présentation de la librairie requests
+Date: 2019-01-31 16:00
+Modified: 2019-01-31 16:00
+Category: Tutoriels
+Tags: requests
+Slug: presentation-requests
+Author: Cédric Migazzi
+
+# Présentation de la librairie requests.
```
[the remark this answers is not in the document]
Comment (cmigazzi) [reply content/articles/presentation-requests.md]: When the title is the same of the file name, we don't need to precise title ?

[the remark this answers is not in the document]
Comment (freezed) [reply content/articles/presentation-requests.md]: `Title` is apart from the `Slug` & `filename`.

Setting a `Slug` in header is useful if the default generation by _Pelican_ do not satisfy or when you have translated versions of the article to have a unique identifier for all translations.

`Title` is mandatory, and in the default theme is used to be shown in top of an article, then in this file, line 9 is a duplicate title

[the remark this answers is not in the document]
Comment (cmigazzi) [reply content/articles/presentation-requests.md]: Ok I got it !

**Pull request closed without merging**